addon/resolver.js

```diff
@@ -105,4 +118,17 @@ export default EmberResolver.extend({
 
     return false;
   },
+
+  /**
+    Checks `type` need use prefix received through a {{#crossLink "DeviceService"}}{{/crossLink}} to lookup class this `type` or not.
+
+    @method _resolveTypeWithDeviceTypeDetection
+    @param {String} type
+    @return {Boolean}
+    @private
+  */
+  _resolveTypeWithDeviceTypeDetection(type) {
+    let resolvingTypes = this.get('resolvingTypes');
+    return resolvingTypes.indexOf(type) > -1;
```

Comment:
Property resolvingTypes must be checked for being an array (Ember.isArray(resolvingTypes)), becouse it can be overridden in the application.